cas-server-core/src/main/java/org/jasig/cas/ticket/TicketGrantingTicketImpl.java

```diff
@@ -77,19 +82,24 @@ public TicketGrantingTicketImpl() {
      * Constructs a new TicketGrantingTicket.
      * 
      * @param id the id of the Ticket
+     * @param proxiedBy Service that produced this proxy ticket.
      * @param ticketGrantingTicket the parent ticket
      * @param authentication the Authentication request for this ticket
      * @param policy the expiration policy for this ticket.
      * @throws IllegalArgumentException if the Authentication object is null
      */
     public TicketGrantingTicketImpl(final String id,
-        final TicketGrantingTicketImpl ticketGrantingTicket,
+        final String proxiedBy, final TicketGrantingTicketImpl ticketGrantingTicket,
```

Comment:
BTW, shouldn't we rename `ticketGrantingTicket` as `parentTGT` or something to make that obvious?